ValveResourceFormat/KeyValues/KeyValues3.cs

```diff
@@ -0,0 +1,535 @@
+
+                //Can directly be added
+                parser.objStack.Peek().AddProperty(parser.currentName, new KVValue(KVType.BOOLEAN, false));
+
+                //Skip next characters
+                SkipChars(parser, "false".Length - 1);
+            }
+            //Boolean true
+            else if (ReadAheadMatches(parser, c, "true"))
+            {
+                parser.stateStack.Pop();
+
+                //Can directly be added
+                parser.objStack.Peek().AddProperty(parser.currentName, new KVValue(KVType.BOOLEAN, true));
+
+                //Skip next characters
+                SkipChars(parser, "true".Length - 1);
+            }
+            //Number
+            else if (numerals.Contains(c))
+            {
+                parser.stateStack.Pop();
+                parser.stateStack.Push(State.VALUE_NUMBER);
+                parser.currentString.Clear();
+                parser.currentString.Append(c);
+            }
+            //Flagged resource
+            else
+            {
+                parser.stateStack.Pop();
+                parser.stateStack.Push(State.VALUE_FLAGGED);
+                parser.currentString.Clear();
+                parser.currentString.Append(c);
+            }
+        }
+
+        //Reading a property name
+        private static void ReadPropName(char c, Parser parser)
+        {
+            //Stop once whitespace is encountered
+            if (Char.IsWhiteSpace(c))
+            {
+                parser.stateStack.Pop();
+                parser.stateStack.Push(State.SEEK_VALUE);
+                parser.currentName = parser.currentString.ToString();
+                return;
+            }
+
+            parser.currentString.Append(c);
+        }
+
+        //Read a structure
+        private static void ReadValueStruct(char c, Parser parser)
+        {
+            //Ignore whitespace
+            if (Char.IsWhiteSpace(c))
+            {
+                return;
+            }
+
+            //Catch comments
+            if (c == '/')
+            {
+                parser.stateStack.Push(State.COMMENT);
+                parser.currentString.Clear();
+                parser.currentString.Append(c);
+                return;
+            }
+
+            //Check for the end of the structure
+            if (c == '}')
+            {
+                KVObject value = parser.objStack.Pop();
+                parser.objStack.Peek().AddProperty(value.Key, new KVValue(KVType.OBJECT, value));
+                parser.stateStack.Pop();
+                return;
+            }
+
+            //Start looking for the next property name
+            parser.stateStack.Push(State.PROP_NAME);
+            parser.currentString.Clear();
+            parser.currentString.Append(c);
+        }
+
+        //Read a string value
+        private static void ReadValueString(char c, Parser parser)
+        {
+            if (c == '"' && parser.previousChar != '\\')
```

Comment:
How does Valve's parser handle escape sequences for chars other than `"`? What would happen if it's `\0` or `\40`?